File: src/jvmMain/java/core/FileUtils.java
package core;

/* Simone 04/12/13 11.59 */

import java.io.*;
import java.nio.charset.Charset;
import java.nio.file.CopyOption;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Collection;
import java.util.List;

public class FileUtils {
    public static void write(String file, CharSequence data) {
        File file1 = new File(file);
        write(file1, data);
    }

    public static void write(String file, byte[] data) {
        File file1 = new File(file);
        write(file1, data);
    }

    public static void write(Path path, CharSequence data) {
        write(path.toFile(), data);
    }

    public static void write(Path path, byte[] data) {
        write(path.toFile(), data);
    }

    public static void append(Path path, CharSequence data) {
        append(path.toFile(), data);
    }

    public static void append(String file, CharSequence data) {
        File file1 = new File(file);
        append(file1, data);
    }

    public static void append(File file, CharSequence data) {
        try {
            org.apache.commons.io.FileUtils.write(file, data, Charset.defaultCharset(), true);
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static void append(File file, byte[] data) {
        try {
            try (FileOutputStream fos = new FileOutputStream(file, true)) {
                fos.write(data);
            }
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static void append(String path, byte[] data) {
        try {
            try (FileOutputStream fos = new FileOutputStream(new File(path), true)) {
                fos.write(data);
            }
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static void write(File file, CharSequence data) {
        try {
            org.apache.commons.io.FileUtils.write(file, data, (String) null, false);
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static void write(File file, byte[] data) {
        try {
            FileOutputStream fos = new FileOutputStream(file);
            try {
                fos.write(data);
            } finally {
                fos.close();
            }
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static String read(File file) {
        try {
            return org.apache.commons.io.FileUtils.readFileToString(file, Charset.defaultCharset());
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static String read(String path) {
        return read(new File(path));
    }

    public static List<String> readLines(String path) {
        return readLines(new File(path));
    }

    public static List<String> readLines(File file) {
        try {
            return org.apache.commons.io.FileUtils.readLines(file);
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static List<String> readLines(Path path) {
        return readLines(path.toFile());
    }

    public static String read(Path path) {
        try {
            return org.apache.commons.io.FileUtils.readFileToString(path.toFile(), Charset.defaultCharset());
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static void writeLines(File file, Collection<?> lines) {
        try {
            org.apache.commons.io.FileUtils.writeLines(file, lines);
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }


    public static FileOutputStream newFileOutputStream(File file) {
        try {
            return new FileOutputStream(file);
        } catch (FileNotFoundException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static FileInputStream newFileInputStream(File file) {
        try {
            return new FileInputStream(file);
        } catch (FileNotFoundException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static Path move(Path source, Path target, CopyOption... options) {
        try {
            return Files.move(source, target, options);
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static void delete(File file) {
        if (file.exists())
            file.delete();
    }

    public static boolean createNewFile(File file) {
        try {
            return file.createNewFile();
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static byte[] readBytes(File file) {
        try {
            return org.apache.commons.io.FileUtils.readFileToByteArray(file);
        } catch (IOException e) {
            throw new RuntimeIOException(e);
        }
    }

    public static byte[] readBytes(Path path) {
        return readBytes(path.toFile());
    }

    public static byte[] readBytes(String path) {
        return readBytes(new File(path));
    }

    public static void delete(String path) {
        delete(new File(path));
    }
}

File: satd-classifier/src/jvmMain/kotlin/core/FileUtils.kt
package core

import org.apache.commons.io.FileUtils
import java.io.*
import java.nio.charset.Charset
import java.nio.file.CopyOption
import java.nio.file.Files
import java.nio.file.Path

/* Simone 04/12/13 11.59 */   object FileUtils {
    fun write(file: String?, data: CharSequence?) {
        val file1 = File(file)
        write(file1, data)
    }

    fun write(file: String?, data: ByteArray?) {
        val file1 = File(file)
        write(file1, data)
    }

    fun write(path: Path, data: CharSequence?) {
        write(path.toFile(), data)
    }

    fun write(path: Path, data: ByteArray?) {
        write(path.toFile(), data)
    }

    fun append(path: Path, data: CharSequence?) {
        append(path.toFile(), data)
    }

    fun append(file: String?, data: CharSequence?) {
        val file1 = File(file)
        append(file1, data)
    }

    fun append(file: File?, data: CharSequence?) {
        try {
            FileUtils.write(file, data, Charset.defaultCharset(), true)
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun append(file: File?, data: ByteArray?) {
        try {
            FileOutputStream(file, true).use { fos -> fos.write(data) }
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun append(path: String?, data: ByteArray?) {
        try {
            FileOutputStream(File(path), true).use { fos -> fos.write(data) }
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun write(file: File?, data: CharSequence?) {
        try {
            FileUtils.write(file, data, null as String?, false)
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun write(file: File?, data: ByteArray?) {
        try {
            val fos = FileOutputStream(file)
            try {
                fos.write(data)
            } finally {
                fos.close()
            }
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun read(file: File?): String {
        return try {
            FileUtils.readFileToString(file, Charset.defaultCharset())
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun read(path: String?): String {
        return read(File(path))
    }

    fun readLines(path: String?): List<String> {
        return readLines(File(path))
    }

    fun readLines(file: File?): List<String> {
        return try {
            FileUtils.readLines(file)
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun readLines(path: Path): List<String> {
        return readLines(path.toFile())
    }

    fun read(path: Path): String {
        return try {
            FileUtils.readFileToString(path.toFile(), Charset.defaultCharset())
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun writeLines(file: File?, lines: Collection<*>?) {
        try {
            FileUtils.writeLines(file, lines)
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun newFileOutputStream(file: File?): FileOutputStream {
        return try {
            FileOutputStream(file)
        } catch (e: FileNotFoundException) {
            throw RuntimeIOException(e)
        }
    }

    fun newFileInputStream(file: File?): FileInputStream {
        return try {
            FileInputStream(file)
        } catch (e: FileNotFoundException) {
            throw RuntimeIOException(e)
        }
    }

    fun move(
        source: Path?,
        target: Path?,
        vararg options: CopyOption?
    ): Path {
        return try {
            Files.move(source, target, *options)
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun delete(file: File) {
        if (file.exists()) file.delete()
    }

    fun createNewFile(file: File): Boolean {
        return try {
            file.createNewFile()
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun readBytes(file: File?): ByteArray {
        return try {
            FileUtils.readFileToByteArray(file)
        } catch (e: IOException) {
            throw RuntimeIOException(e)
        }
    }

    fun readBytes(path: Path): ByteArray {
        return readBytes(path.toFile())
    }

    fun readBytes(path: String?): ByteArray {
        return readBytes(File(path))
    }

    fun delete(path: String?) {
        delete(File(path))
    }
}